doc/CMakeLists.txt

```diff
@@ -153,52 +153,21 @@ add_custom_target(
 
 set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${GEN_KCONFIG_REST_SCRIPT})
 
-#
-# Generated devicetree .rst documents
-#
-# The devicetree bindings discovered in ${DTS_ROOTS} are parsed and
-# documentation for them is generated in the directory
-# ${DTS_BINDINGS_RST_OUT}.
-#
-# The CMake variable GEN_DEVICETREE_REST_ZEPHYR_DOCSET will
-# be passed to the script in the environment. This allows separating
-# the bindings documentation into a standalone Sphinx docset that
-# nonetheless can link to Zephyr documentation using intersphinx.
-# If empty, the variable has no effect on the script.
-#
-# If not set, these are the default values for these variables:
-#
-# - DTS_ROOTS: ZEPHYR_BASE
-# - DTS_BINDINGS_RST_OUT: ${DOCS_SRC_DIR}/reference/devicetree
-#
+#-------------------------------------------------------------------------------
+# devicetree
 
 set(GEN_DEVICETREE_REST_SCRIPT ${CMAKE_CURRENT_LIST_DIR}/_scripts/gen_devicetree_rest.py)
 
-if(NOT DTS_ROOTS)
-  set(DTS_ROOTS ${ZEPHYR_BASE})
-endif()
-
-set(DTS_ROOT_ARGS)
-foreach(root ${DTS_ROOTS})
-  list(APPEND DTS_ROOT_ARGS --dts-root ${root})
-endforeach()
-
-if(NOT DTS_BINDINGS_RST_OUT)
-  set(DTS_BINDINGS_RST_OUT ${DOCS_SRC_DIR}/reference/devicetree)
-endif()
-
 add_custom_target(
   devicetree
   COMMAND ${CMAKE_COMMAND} -E env
   PYTHONPATH=${ZEPHYR_BASE}/scripts/dts/python-devicetree/src${SEP}$ENV{PYTHONPATH}
   ZEPHYR_BASE=${ZEPHYR_BASE}
-  GEN_DEVICETREE_REST_ZEPHYR_DOCSET=${GEN_DEVICETREE_REST_ZEPHYR_DOCSET}
   ${PYTHON_EXECUTABLE} ${GEN_DEVICETREE_REST_SCRIPT}
     --vendor-prefixes ${ZEPHYR_BASE}/dts/bindings/vendor-prefixes.txt
-    ${DTS_ROOT_ARGS} ${DTS_BINDINGS_RST_OUT}
+    --dts-root ${ZEPHYR_BASE}
```

Comment:
Why are you removing the support for out of tree bindings ?
There was a reason this was added:
https://github.com/zephyrproject-rtos/zephyr/issues/28865
https://github.com/zephyrproject-rtos/zephyr/commit/808dd9e6843c72135e2b2bf87d1080519a45ba4e

> Most of the extra options are not
used, so they have been removed to make things easier to maintain.

How do you know ?

Reply:
We never set `DTS_ROOTS` nor `DTS_BINDINGS_RST_OUT` for Zephyr docs. It was probably added for external users (i.e. NCS), but NCS stopped using Zephyr docs CMake.

Reply:
> It was probably added for external users (i.e. NCS), but NCS stopped using Zephyr docs CMake.

But someone else could be using it.
So we should not remove such feature as long as we don't know if it is being used by others.

If we want to remove a feature, then it should be deprecated first, to give downstream users time to adjust / react.

Reply:
I think these options in particular offer extensibility that the Zephyr docs do permit. For example, changing the RST out folder  may imply changing some toctree to include the index.  Regarding DTS_ROOTS, I think it's something up to Zephyr to decide which folders to document. If we device to include bindings in modules, it's something that should be handled internally and not via an external parameter.

Reply:
> If we device to include bindings in modules, it's something that should be handled internally and not via an external parameter.

At the time https://github.com/zephyrproject-rtos/zephyr/commit/808dd9e6843c72135e2b2bf87d1080519a45ba4e was created the Zephyr modules did not offer `board_root`, `dts_root`, etc in a modules `zephyr/module.yml` file.
https://docs.zephyrproject.org/latest/guides/modules.html#build-settings

but it does so now. So feel free to rework into supporting those directly from the modules if a `dts_root` is given, but I don't see why removing the option is justified.

After all, the DTS_ROOT will only be included in the docset if the caller passes that setting when building the doc.
And if the user is passing this setting, I assume he has a reason for doing so.

Reply:
I've added `DTS_ROOTS` back to at least allow experimenting with external bindings.